### PR TITLE
Moved Litespeed compatibility in OOP Subscriber

### PR DESCRIPTION
**Pull request by crystinutzaa**
Moved Litespeed compatibility in OOP Subscriber

### CODE REVIEW

File: inc/classes/subscriber/third-party/Hostings/class-litespeed-subscriber.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	 * @inheritDoc
+	 */
+	public static function get_subscribed_events() {
+		if ( ! isset( $_SERVER['HTTP_X_LSCACHE'] ) ) {
+			return [];
+		}
+
+		return [
+			'before_rocket_clean_domain' => 'rocket_litespeed_clean_domain',
+			'before_rocket_clean_file'   => [ 'rocket_litespeed_clean_file', 10, 1 ],
```
Comment (Screenfeed) [inline inc/classes/subscriber/third-party/Hostings/class-litespeed-subscriber.php]: `10` and `1` are the default values, so they can be omitted. The line can simply be written like this:
```php
'before_rocket_clean_file'   => 'rocket_litespeed_clean_file',
```
Anyway, it doesn't hurt to keep it like this :)

File: inc/functions/cloudflare.php [hunk 2]
```diff
@@ -14,25 +14,30 @@
...
 
 	if ( ! isset( $cf_email, $cf_api_key ) || empty( $cf_email ) || empty( $cf_api_key ) ) {
-		return new WP_Error( 'cloudflare_credentials_empty', __( 'Cloudflare Email and API key are not set.', 'rocket' ) );
+		return new WP_Error(
+			'cloudflare_credentials_empty',
+			sprintf(
+				/* translators: %1$s = opening link; %2$s = closing link */
+				__( 'Cloudflare email, API key and Zone ID are not set. Read the %1$sdocumentation%2$s for further guidance.', 'rocket' ),
+				// translators: Documentation exists in EN, FR; use localized URL if applicable.
+				'<a href="' . __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) . '" target="_blank">',
```
Comment (Screenfeed) [inline inc/functions/cloudflare.php]: We totally forgot this.

```suggestion
				'<a href="' . esc_url( __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) ) . '" rel="noopener noreferrer" target="_blank">',
```

File: inc/functions/cloudflare.php [hunk 3]
```diff
@@ -14,25 +14,30 @@
...
-		);
+		$msg = __( 'Missing Cloudflare Zone ID.', 'rocket' );
 
 		$msg .= ' ' . sprintf(
 			/* translators: %1$s = opening link; %2$s = closing link */
 			__( 'Read the %1$sdocumentation%2$s for further guidance.', 'rocket' ),
-			/* translators: Documentation exists in EN, DE, FR, ES, IT; use loaclised URL if applicable */
-			'<a href="' . __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket', 'rocket' ) . '" target="_blank">',
+			// translators: Documentation exists in EN, FR; use localized URL if applicable.
+			'<a href="' . __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) . '" target="_blank">',
```
Comment (Screenfeed) [inline inc/functions/cloudflare.php]:
```suggestion
			'<a href="' . esc_url( __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) ) . '" rel="noopener noreferrer" target="_blank">',
```

File: inc/functions/cloudflare.php [hunk 4]
```diff
@@ -41,18 +46,38 @@ function rocket_is_api_keys_valid_cloudflare( $cf_email, $cf_api_key, $cf_zone_i
...
 
 		if ( ! isset( $cf_zone->success ) || empty( $cf_zone->success ) ) {
-			return new WP_Error( 'cloudflare_invalid_auth', $cf_zone->errors[0]->message );
+			$msg = __( 'Incorrect Cloudflare Zone ID.', 'rocket' );
+
+			$msg .= ' ' . sprintf(
+				/* translators: %1$s = opening link; %2$s = closing link */
+				__( 'Read the %1$sdocumentation%2$s for further guidance.', 'rocket' ),
+				// translators: Documentation exists in EN, FR; use localized URL if applicable.
+				'<a href="' . __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) . '" target="_blank">',
```
Comment (Screenfeed) [inline inc/functions/cloudflare.php]:
```suggestion
				'<a href="' . esc_url( __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) ) . '" rel="noopener noreferrer" target="_blank">',
```

File: inc/functions/cloudflare.php [hunk 5]
```diff
@@ -41,18 +46,38 @@ function rocket_is_api_keys_valid_cloudflare( $cf_email, $cf_api_key, $cf_zone_i
...
+
 	} catch ( Exception $e ) {
-		return new WP_Error( 'cloudflare_invalid_auth', $e->getMessage() );
+		$msg = __( 'Incorrect Cloudflare email address or API key.', 'rocket' );
+
+		$msg .= ' ' . sprintf(
+			/* translators: %1$s = opening link; %2$s = closing link */
+			__( 'Read the %1$sdocumentation%2$s for further guidance.', 'rocket' ),
+			// translators: Documentation exists in EN, FR; use localized URL if applicable.
+			'<a href="' . __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) . '" target="_blank">',
```
Comment (Screenfeed) [inline inc/functions/cloudflare.php]:
```suggestion
			'<a href="' . esc_url( __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) ) . '" rel="noopener noreferrer" target="_blank">',
```

File: inc/functions/cloudflare.php [hunk 6]
```diff
@@ -66,14 +91,23 @@ function rocket_is_api_keys_valid_cloudflare( $cf_email, $cf_api_key, $cf_zone_i
...
 
 	if ( ! isset( $cf_email, $cf_api_key ) ) {
-		return new WP_Error( 'cloudflare_credentials_empty', __( 'Cloudflare Email and API key are not set', 'rocket' ) );
+		$msg = __( 'Cloudflare email and API key are not set.', 'rocket' );
+
+		$msg .= ' ' . sprintf(
+			/* translators: %1$s = opening link; %2$s = closing link */
+			__( 'Read the %1$sdocumentation%2$s for further guidance.', 'rocket' ),
+			// translators: Documentation exists in EN, FR; use localized URL if applicable.
+			'<a href="' . __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) . '" target="_blank">',
```
Comment (Screenfeed) [inline inc/functions/cloudflare.php]:
```suggestion
			'<a href="' . esc_url( __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) ) . '" rel="noopener noreferrer" target="_blank">',
```

File: inc/functions/cloudflare.php [hunk 7]
```diff
@@ -102,17 +136,13 @@ function get_rocket_cloudflare_instance() {
...
-		);
+		$msg = __( 'Missing Cloudflare Zone ID.', 'rocket' );
 
 		$msg .= ' ' . sprintf(
 			/* translators: %1$s = opening link; %2$s = closing link */
 			__( 'Read the %1$sdocumentation%2$s for further guidance.', 'rocket' ),
-			/* translators: Documentation exists in EN, DE, FR, ES, IT; use loaclised URL if applicable */
-			'<a href="' . __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket', 'rocket' ) . '" target="_blank">',
+			// translators: Documentation exists in EN, FR; use localized URL if applicable.
+			'<a href="' . __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) . '" target="_blank">',
```
Comment (Screenfeed) [inline inc/functions/cloudflare.php]:
```suggestion
			'<a href="' . esc_url( __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) ) . '" rel="noopener noreferrer" target="_blank">',
```

File: inc/functions/cloudflare.php [hunk 8]
```diff
@@ -448,18 +478,31 @@ function rocket_purge_cloudflare() {
...
-
-			$errors = implode( ', ', $errors );
-			throw new Exception( $errors );
+			$msg = __( 'Incorrect Cloudflare Zone ID.', 'rocket' );
+
+			$msg .= ' ' . sprintf(
+				/* translators: %1$s = opening link; %2$s = closing link */
+				__( 'Read the %1$sdocumentation%2$s for further guidance.', 'rocket' ),
+				// translators: Documentation exists in EN, FR; use localized URL if applicable.
+				'<a href="' . __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) . '" target="_blank">',
```
Comment (Screenfeed) [inline inc/functions/cloudflare.php]:
```suggestion
				'<a href="' . esc_url( __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) ) . '" rel="noopener noreferrer" target="_blank">',
```

File: inc/functions/cloudflare.php [hunk 9]
```diff
@@ -448,18 +478,31 @@ function rocket_purge_cloudflare() {
...
 
 	} catch ( Exception $e ) {
-		return new WP_Error( 'cloudflare_purge_failed', $e->getMessage() );
+		$msg = __( 'Incorrect Cloudflare email address or API key.', 'rocket' );
+
+		$msg .= ' ' . sprintf(
+			/* translators: %1$s = opening link; %2$s = closing link */
+			__( 'Read the %1$sdocumentation%2$s for further guidance.', 'rocket' ),
+			// translators: Documentation exists in EN, FR; use localized URL if applicable.
+			'<a href="' . __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) . '" target="_blank">',
```
Comment (Screenfeed) [inline inc/functions/cloudflare.php]:
```suggestion
			'<a href="' . esc_url( __( 'https://docs.wp-rocket.me/article/18-using-wp-rocket-with-cloudflare/?utm_source=wp_plugin&utm_medium=wp_rocket#add-on', 'rocket' ) ) . '" rel="noopener noreferrer" target="_blank">',
```

File: inc/classes/subscriber/third-party/Hostings/class-litespeed-subscriber.php (hunk 1 above)
Comment (crystinutzaa) [reply inc/classes/subscriber/third-party/Hostings/class-litespeed-subscriber.php]: Removed default values

**Pull request merged**